pkg/types/datum.go

```diff
@@ -1012,9 +1012,6 @@ func ProduceFloatWithSpecifiedTp(f float64, target *FieldType, sc *stmtctx.State
 	// If no D is set, we will handle it like origin float whether M is set or not.
 	if target.GetFlen() != UnspecifiedLength && target.GetDecimal() != UnspecifiedLength {
 		f, err = TruncateFloat(f, target.GetFlen(), target.GetDecimal())
-		if err = sc.HandleOverflow(err, err); err != nil {
```

Comment:
`HandleOverflow` seems useless now, because in cast expr like `cast("1" as float(2, 1))` , `target.GetDecimal()` always return `UnspecifiedLength` ...

Reply:
In mysql cast to float with decimal will fail:

```
mysql> select cast("1" as float(1, 0));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ', 0))' at line 1
```